### ISSUE TITLE
Internal Compiler Error: sentence-transformer with neuroncore_pipeline_cores = 1 and batch = 6

### ISSUE THREAD

**Issue by Matthieu-Tinycoaching** [aws/aws-neuron-sdk#282]
```
Libraries versions:
neuron-cc == 1.4.1.0
torch-neuron == 1.8.1.1.4.1.0
```

Hello,

I tried to compile a sentence-transformer model with `neuroncore_pipeline_cores = 1` and `batch = 6` according the following code:

```
import tensorflow  # to workaround a protobuf version conflict issue
import torch
import torch.neuron
import transformers

#### I/ Compile the sts-transformer with torchscript ####

model_name = "sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2"
pipeline_name = "feature-extraction"
nlp = transformers.pipeline(pipeline_name, model=model_name, tokenizer=model_name, device=-1)
tokenizer = nlp.tokenizer

class SentenceTransformerCustom(transformers.BertModel):
    def __init__(self, config):
        super().__init__(config)
        # Naming alias for ONNX output specification
        # Makes it easier to identify the layer
        self.sentence_embedding = torch.nn.Identity()

    def forward(self, input_ids, attention_mask):
        # Get the token embeddings from the base model
        token_embeddings = super().forward(
            input_ids, 
            attention_mask=attention_mask
        )[0]
        # Stack the pooling layer on top of it
        input_mask_expanded = attention_mask.unsqueeze(-1).expand(token_embeddings.size()).float()
        sum_embeddings = torch.sum(token_embeddings * input_mask_expanded, 1)
        sum_mask = torch.clamp(input_mask_expanded.sum(1), min=1e-9)
        return self.sentence_embedding(sum_embeddings / sum_mask)

model = SentenceTransformerCustom(config=nlp.model.config).from_pretrained(model_name, return_dict=False)

src_text = "Adresse électronique : Pour pouvoir s'échanger des messages électroniques il faut une adresse... électronique ! Tout comme avoir une adresse postale est mieux pour recevoir une lettre. L'adresse électronique peut être obtenue auprès d'un fournisseur de messagerie (Gmail, Hotmail, etc.), et se découpe en 3 parties : l'identifiant que vous avez choisi (ex. : bobforever), l'arobase (@), et le nom de domaine (ex. : gmail.com)."
max_length=128
dummy_phrase = tokenizer(src_text, padding='max_length', truncation=True, max_length=max_length, return_tensors='pt')

batch_size = 6
example_inputs_dummy_phrase = (
    torch.cat([dummy_phrase['input_ids']] * batch_size,0),
    torch.cat([dummy_phrase['attention_mask']] * batch_size,0)
)

neuron_model = torch.neuron.trace(model, 
                                example_inputs = example_inputs_dummy_phrase,
                                verbose=1,
                                compiler_workdir='./'
                                )
neuron_model.save("paraphrase-multilingual-MiniLM-L12-v2-custom-neuron-data-parallel-pad128-b{}.pt".format(batch_size))
```

but got the following error log:

```
06/16/2021 04:16:55 PM INFO [WalrusDriver]: Walrus pass: pre_sched succeeded!
06/16/2021 04:16:55 PM INFO [WalrusDriver]: Walrus pass: birverifier succeeded!
06/16/2021 04:16:55 PM INFO [WalrusDriver]: Running walrus pass: coloring_allocator
06/16/2021 04:17:15 PM INFO [WalrusDriver]: INFO: GCA DRAM bytes loaded or saved 45090304
06/16/2021 04:17:15 PM INFO [WalrusDriver]: INFO: GCA average DMA size 2603 bytes
06/16/2021 04:17:15 PM INFO [WalrusDriver]: Allocating functions
06/16/2021 04:17:15 PM INFO [WalrusDriver]:   linearize and check
06/16/2021 04:17:15 PM CRITICAL [WalrusDriver]: Bad input file - failed static checks.
06/16/2021 04:17:15 PM CRITICAL [WalrusDriver]:   12 names were undefined at use point
06/16/2021 04:17:15 PM CRITICAL [WalrusDriver]:     t13061_i1
06/16/2021 04:17:15 PM CRITICAL [WalrusDriver]:     t13061_i2
06/16/2021 04:17:15 PM CRITICAL [WalrusDriver]:     t13061_i5
06/16/2021 04:17:15 PM CRITICAL [WalrusDriver]:     t13061_i6
06/16/2021 04:17:15 PM CRITICAL [WalrusDriver]:     t13061_i9
06/16/2021 04:17:15 PM CRITICAL [WalrusDriver]:     t13061_i10
06/16/2021 04:17:15 PM CRITICAL [WalrusDriver]:       etc.
Aborted (core dumped)
06/16/2021 04:17:15 PM ERROR [neuron-cc]: ***************************************************************
06/16/2021 04:17:15 PM ERROR [neuron-cc]:  An Internal Compiler Error has occurred
06/16/2021 04:17:15 PM ERROR [neuron-cc]: ***************************************************************
06/16/2021 04:17:15 PM ERROR [neuron-cc]: 
06/16/2021 04:17:15 PM ERROR [neuron-cc]: Error message:  Non-zero exit status (134) for command: /home/matthieu/neuron_fresh_venv/lib/python3.6/site-packages/neuroncc/starfish/bin/walrus_driver --optlevel=2 --verbose=20 -o walrus_bir.out.json -i bir.json  --allocator=coloring --min_split_size=10240 --max-partitions=1 --policy=2 --latency=6  --internal-hyper-parameters=/opt/ml/input/config/hyperparameters.json --tuning=2 --numcores=1
06/16/2021 04:17:15 PM ERROR [neuron-cc]: 
06/16/2021 04:17:15 PM ERROR [neuron-cc]: Error class:    CompilerInternalError
06/16/2021 04:17:15 PM ERROR [neuron-cc]: Error location: job.WalrusDriver.3
06/16/2021 04:17:15 PM ERROR [neuron-cc]: Command line:   /home/matthieu/neuron_fresh_venv/bin/neuron-cc compile /home/matthieu/Code/74/graph_def.pb --framework TENSORFLOW --pipeline compile SaveTemps --output /home/matthieu/Code/74/graph_def.neff --io-config '{"inputs": {"0:0": [[6, 128, 384], "float32"], "1:0": [[6, 1, 1, 128], "float32"], "2:0": [[6, 128], "int64"]}, "outputs": ["truediv_12:0"]}' --verbose 1
06/16/2021 04:17:15 PM ERROR [neuron-cc]: 
06/16/2021 04:17:15 PM ERROR [neuron-cc]: Internal details:
06/16/2021 04:17:15 PM ERROR [neuron-cc]:   File "neuroncc/driver/Job.py", line 244, in neuroncc.driver.Job.runSingleInputFn
06/16/2021 04:17:15 PM ERROR [neuron-cc]:   File "neuroncc/driver/jobs/WalrusDriver.py", line 228, in neuroncc.driver.jobs.WalrusDriver.WalrusDriver.runSingleInput
06/16/2021 04:17:15 PM ERROR [neuron-cc]:   File "neuroncc/driver/Job.py", line 178, in neuroncc.driver.Job.Job.shellCommand
06/16/2021 04:17:15 PM ERROR [neuron-cc]: 
06/16/2021 04:17:15 PM ERROR [neuron-cc]: Version information:
06/16/2021 04:17:15 PM ERROR [neuron-cc]:   Neuron Compiler version 1.4.1.0+737cbb69a
06/16/2021 04:17:15 PM ERROR [neuron-cc]:   
06/16/2021 04:17:15 PM ERROR [neuron-cc]:   HWM version 1.4.1.0-0
06/16/2021 04:17:15 PM ERROR [neuron-cc]:   NEFF version 1.0
06/16/2021 04:17:15 PM ERROR [neuron-cc]:   TVM version 1.4.1.0+0
06/16/2021 04:17:15 PM ERROR [neuron-cc]:   NumPy version 1.19.5
06/16/2021 04:17:15 PM ERROR [neuron-cc]:   MXNet not available
06/16/2021 04:17:15 PM ERROR [neuron-cc]:   TF not available
06/16/2021 04:17:15 PM ERROR [neuron-cc]:   ONNX not available
06/16/2021 04:17:15 PM ERROR [neuron-cc]: 
06/16/2021 04:17:15 PM ERROR [neuron-cc]: Artifacts stored in: /home/matthieu/Code/74
INFO:Neuron:Compile command returned: 1
WARNING:Neuron:torch.neuron.trace failed on _NeuronGraph$655; falling back to native python function call
ERROR:Neuron:neuron-cc failed with the following command line call:
/home/matthieu/neuron_fresh_venv/bin/neuron-cc compile /home/matthieu/Code/74/graph_def.pb --framework TENSORFLOW --pipeline compile SaveTemps --output /home/matthieu/Code/74/graph_def.neff --io-config '{"inputs": {"0:0": [[6, 128, 384], "float32"], "1:0": [[6, 1, 1, 128], "float32"], "2:0": [[6, 128], "int64"]}, "outputs": ["truediv_12:0"]}' --verbose 1
Traceback (most recent call last):
  File "/home/matthieu/neuron_fresh_venv/lib/python3.6/site-packages/torch_neuron/convert.py", line 330, in op_converter
    dynamic_batch_size=self.dynamic_batch_size, **kwargs)
  File "/home/matthieu/neuron_fresh_venv/lib/python3.6/site-packages/torch_neuron/decorators.py", line 191, in trace
    'neuron-cc failed with the following command line call:\n{}'.format(command))
subprocess.SubprocessError: neuron-cc failed with the following command line call:
/home/matthieu/neuron_fresh_venv/bin/neuron-cc compile /home/matthieu/Code/74/graph_def.pb --framework TENSORFLOW --pipeline compile SaveTemps --output /home/matthieu/Code/74/graph_def.neff --io-config '{"inputs": {"0:0": [[6, 128, 384], "float32"], "1:0": [[6, 1, 1, 128], "float32"], "2:0": [[6, 128], "int64"]}, "outputs": ["truediv_12:0"]}' --verbose 1
INFO:Neuron:Number of arithmetic operators (post-compilation) before = 726, compiled = 0, percent compiled = 0.0%
INFO:Neuron:The neuron partitioner created 1 sub-graphs
INFO:Neuron:Neuron successfully compiled 0 sub-graphs, Total fused subgraphs = 1, Percent of model sub-graphs successfully compiled = 0.0%
INFO:Neuron:Compiled these operators (and operator counts) to Neuron:
INFO:Neuron:Not compiled operators (and operator counts) to Neuron:
INFO:Neuron: => aten::Int: 102 [supported]
INFO:Neuron: => aten::add: 111 [supported]
INFO:Neuron: => aten::addmm: 1 [supported]
INFO:Neuron: => aten::clamp: 1 [supported]
INFO:Neuron: => aten::contiguous: 12 [supported]
INFO:Neuron: => aten::div: 13 [supported]
INFO:Neuron: => aten::dropout: 37 [supported]
INFO:Neuron: => aten::embedding: 3 [not supported]
INFO:Neuron: => aten::expand: 1 [supported]
INFO:Neuron: => aten::gelu: 12 [supported]
INFO:Neuron: => aten::layer_norm: 25 [supported]
INFO:Neuron: => aten::matmul: 96 [supported]
INFO:Neuron: => aten::mul: 2 [supported]
INFO:Neuron: => aten::permute: 48 [supported]
INFO:Neuron: => aten::rsub: 1 [supported]
INFO:Neuron: => aten::select: 1 [supported]
INFO:Neuron: => aten::size: 102 [supported]
INFO:Neuron: => aten::slice: 5 [supported]
INFO:Neuron: => aten::softmax: 12 [supported]
INFO:Neuron: => aten::sum: 2 [supported]
INFO:Neuron: => aten::t: 73 [supported]
INFO:Neuron: => aten::to: 2 [supported]
INFO:Neuron: => aten::transpose: 12 [supported]
INFO:Neuron: => aten::unsqueeze: 3 [supported]
INFO:Neuron: => aten::view: 48 [supported]
INFO:Neuron: => aten::zeros: 1 [supported]
Traceback (most recent call last):
  File "/home/matthieu/Code/Python/torchserve_sts_transformer_torchscript_cpu_pad128_b1.py", line 83, in <module>
    compiler_workdir='./'
  File "/home/matthieu/neuron_fresh_venv/lib/python3.6/site-packages/torch_neuron/convert.py", line 108, in trace
    cu.stats_post_compiler(neuron_graph)
  File "/home/matthieu/neuron_fresh_venv/lib/python3.6/site-packages/torch_neuron/convert.py", line 441, in stats_post_compiler
    "No operations were successfully partitioned and compiled to neuron for this model - aborting trace!")
RuntimeError: No operations were successfully partitioned and compiled to neuron for this model - aborting trace!
```

Please find the archive of compiled artifacts directory at aws-neuron-support@amazon.com for further debug.

Thanks!

**Comment by jeffhataws** [comment]
Thanks for the artifacts. We will take a look.

**Comment by mrnikwaws** [comment]
Hi Matthieu,

We have identified the issue and will release the fix with an upcoming release.

**Comment by aws-taylor** [comment]
Hello @Matthieu-Tinycoaching,

 This issue has been resolved in our [1.14.1 release](https://awsdocs-neuron.readthedocs-hosted.com/en/latest/release-notes/releasecontent.html#current-release-neuron-v1-14-1-july-2nd-2021). If you have any further issues please let us know.

Regards,
Taylor
